--- a/arch/arm/boot/dts/sun8i-h3-orangepi-one.dts	2016-08-24 21:07:27.949540293 +1000
+++ b/arch/arm/boot/dts/sun8i-h3-orangepi-one.dts	2016-08-24 21:11:30.931552757 +1000
@@ -111,22 +111,23 @@
 	};
 };
 
-&cpu0 {
-	operating-points = <
-		1200000	1300000
-		1008000	1300000
-		816000	1100000
-		624000	1100000
-		480000	1100000
-		312000	1100000
-		240000	1100000
-		120000	1100000
-		>;
-	#cooling-cells = <2>;
-	cooling-min-level = <0>;
-	cooling-max-level = <7>;
-	cpu0-supply = <&vdd_cpux>;
-};
+/*	&cpu0 {
+ *		operating-points = <
+ *			1200000	1300000
+ *			1008000	1300000
+ *			816000	1100000
+ *			624000	1100000
+ *			480000	1100000
+ *			312000	1100000
+ *			240000	1100000
+ *			120000	1100000
+ *			>;
+ *		#cooling-cells = <2>;
+ *		cooling-min-level = <0>;
+ *		cooling-max-level = <7>;
+ *		cpu0-supply = <&vdd_cpux>;
+ *	};
+ */
 
 &cpu_thermal {
 	trips {
@@ -176,6 +177,14 @@
 	status = "okay";
 };
 
+&ehci2 {
+	status = "okay";
+};
+
+&ehci3 {
+	status = "okay";
+};
+
 &emac {
 	phy = <&phy1>;
 	phy-mode = "mii";
@@ -205,6 +214,14 @@
 	status = "okay";
 };
 
+&ohci2 {
+	status = "okay";
+};
+
+&ohci3 {
+	status = "okay";
+};
+
 &pio {
 	leds_opc: led_pins@0 {
 		allwinner,pins = "PA15";
@@ -263,9 +280,15 @@
 	status = "okay";
 };
 
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart1_pins_a>;
+	status = "okay";
+};
+
 &usb_otg {
 	dr_mode = "peripheral";
-	status = "okay";
+	status = "disabled";
 };
 
 &usbphy {
