--- a/arch/arm/boot/dts/sun8i-h3.dtsi	2016-08-24 21:07:27.945540338 +1000
+++ b/arch/arm/boot/dts/sun8i-h3.dtsi	2016-08-24 21:07:27.949540293 +1000
@@ -64,28 +64,28 @@
 			reg = <0>;
 			clocks = <&ccu CLK_CPUX>;
 			clock-latency = <244144>; /* 8 32k periods */
-			clock-frequency = <1200000000>;
+			clock-frequency = <1008000000>;
 		};
 
 		cpu@1 {
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
 			reg = <1>;
-			clock-frequency = <1200000000>;
+			clock-frequency = <1008000000>;
 		};
 
 		cpu@2 {
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
 			reg = <2>;
-			clock-frequency = <1200000000>;
+			clock-frequency = <1008000000>;
 		};
 
 		cpu@3 {
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
 			reg = <3>;
-			clock-frequency = <1200000000>;
+			clock-frequency = <1008000000>;
 		};
 	};
 
