--- a/arch/arm/boot/dts/sun8i-h3-orangepi-one.dts	2016-08-25 20:12:53.313157041 +1000
+++ b/arch/arm/boot/dts/sun8i-h3-orangepi-one.dts	2016-08-25 20:12:53.317156984 +1000
@@ -51,7 +51,7 @@
 
 / {
 	model = "Xunlong Orange Pi One";
-	compatible = "xunlong,orangepi-one", "allwinner,sun8i-h3";
+	compatible = "friendlyarm,nanopi-neo", "xunlong,orangepi-one", "allwinner,sun8i-h3";
 
 	aliases {
 		serial0 = &uart0;
@@ -111,68 +111,19 @@
 	};
 };
 
-&cpu0 {
-	operating-points = <
-		1200000	1300000
-		1008000	1300000
-		816000	1100000
-		624000	1100000
-		480000	1100000
-		312000	1100000
-		240000	1100000
-		120000	1100000
-		>;
-	#cooling-cells = <2>;
-	cooling-min-level = <0>;
-	cooling-max-level = <7>;
-	cpu0-supply = <&vdd_cpux>;
-};
-
-&cpu_thermal {
-	trips {
-		cpu_warm: cpu_warm {
-			temperature = <65000>;
-			hysteresis = <2000>;
-			type = "passive";
-		};
-		cpu_hot: cpu_hot {
-			temperature = <75000>;
-			hysteresis = <2000>;
-			type = "passive";
-		};
-		cpu_very_hot: cpu_very_hot {
-			temperature = <90000>;
-			hysteresis = <2000>;
-			type = "passive";
-		};
-		cpu_crit: cpu_crit {
-			temperature = <105000>;
-			hysteresis = <2000>;
-			type = "critical";
-		};
-	};
+&ehci0 {
+	status = "okay";
+};
 
-	cooling-maps {
-		cpu_warm_limit_cpu {
-			trip = <&cpu_warm>;
-			cooling-device = <&cpu0 THERMAL_NO_LIMIT 2>;
-		};
-		cpu_hot_limit_cpu {
-			trip = <&cpu_hot>;
-			cooling-device = <&cpu0 2 4>;
-		};
-		cpu_very_hot_limit_cpu {
-			trip = <&cpu_very_hot>;
-			cooling-device = <&cpu0 6 THERMAL_NO_LIMIT>;
-		};
-	};
+&ehci1 {
+	status = "okay";
 };
 
-&ehci0 {
+&ehci2 {
 	status = "okay";
 };
 
-&ehci1 {
+&ehci3 {
 	status = "okay";
 };
 
@@ -205,6 +156,14 @@
 	status = "okay";
 };
 
+&ohci2 {
+	status = "okay";
+};
+
+&ohci3 {
+	status = "okay";
+};
+
 &pio {
 	leds_opc: led_pins@0 {
 		allwinner,pins = "PA15";
@@ -263,9 +222,26 @@
 	status = "okay";
 };
 
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart1_pins_a>;
+	status = "okay";
+};
+
+&i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c0_pins_a>;
+	status = "okay";
+};
+
+&r_i2c {
+	pinctrl-names = "default";
+	pinctrl-0 = <&r_i2c_pins_a>;
+	status = "okay";
+};
 &usb_otg {
 	dr_mode = "otg";
-	status = "okay";
+	status = "disabled";
 };
 
 &usbphy {
